### PR TITLE
Sends auction_id as query parameter to http solvers.

### PR DESCRIPTION
**Pull request by marcovc**

### CODE REVIEW

File: crates/shared/src/http_solver.rs
Comment (nlordell) [inline crates/shared/src/http_solver.rs:111]: Does the MIP solver support this parameter?

Comment (marcovc) [reply crates/shared/src/http_solver.rs:111]: Not MIP (which is being deprecated btw), not sure about cowdex.

Comment (marcovc) [reply crates/shared/src/http_solver.rs:111]: Can we merge this?

File: crates/shared/src/http_solver.rs
Comment (MartinquaXD) [inline crates/shared/src/http_solver.rs:86]: nit: The value of `maybe_auction_id` can depend on the implementation of `MetadataModel::default()` which is probably not intended. As an alternative you could try this:
```rust
        let auction_id = match &model.metadata {
            Some(metadata) => metadata.auction_id.unwrap_or(0),
            None => 0,
        };
```
I believe this makes a little more obvious what's going on.
Alternatively if you actually want this value to be an `Option<u64>`  this would also work:
```rust 
        let maybe_auction_id = model.metadata.as_ref().and_then(|data| data.auction_id);
```

File: crates/shared/src/http_solver.rs
Comment (MartinquaXD) [reply crates/shared/src/http_solver.rs:111]: Should this also use the same fallback value for `auction_id` as the instance name so that the query parameter always gets added?

Comment (josojo) [reply crates/shared/src/http_solver.rs:111]: Cow does not yet. But after I get an approval in this PR, it should be good: 
https://github.com/gnosis/cow-dex-solver/pull/18

**Comment by marcovc** [comment]
> Should this also use the same fallback value for auction_id as the instance name so that the query parameter always gets added?

At least quasimodo is counting for the presence/absence of auction_id to decide which log info should be appended - if not present then it records an internal request counter. So I'd say to keep it like this if all agree.

File: crates/shared/src/http_solver.rs
Comment (marcovc) [reply crates/shared/src/http_solver.rs:86]: I'll go with the last one, thank you!

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**
